File: LmsPlusPlus.Api/Infrastructure/Task.cs
namespace LmsPlusPlus.Api.Infrastructure;

public record Task
{
    public long Id { get; set; }
    public string Title { get; set; } = null!;
    public string Description { get; set; } = null!;
    public long TopicId { get; set; }
    public Topic Topic { get; set; } = null!;
    public ICollection<Technology> Technologies { get; set; } = null!;
}

File: LmsPlusPlus.Client/src/domain/Task.ts
import { Topic } from "./Topic"
import { Solution } from "./Solution"
import { Technology } from "./Technology"

export class Task {
    static readonly NO_ID = -1
    readonly id: number
    readonly topic: Topic
    readonly title: string
    readonly description: string
    readonly technologies: readonly Technology[]
    private _solutions: Solution[] = []

    get solutions(): Solution[] { return this._solutions }
    set solutions(solutions: Solution[]) { this._solutions = solutions }

    constructor(id: number, topic: Topic, title: string, description: string, technologies: readonly Technology[]) {
        this.id = id
        this.topic = topic
        this.title = title
        this.description = description
        this.technologies = technologies
    }
}
